add merge behavior

mmaelicke opened the pull request:
This PR closes #156 

Now the `ImmutableResultSet.get_data` method automatically merges Composite datasets. For non-compisites, the merge can be requested by the `merge=True` attribute.

In a first test it worked great for the Eddy composite:

![image](https://user-images.githubusercontent.com/2826034/123386254-c0b79700-d596-11eb-98e4-b25ebb1e6de0.png)

Note the shape of the output DataFrame.

@AlexDo1 can you pull this branch and run it on the updated Eddy branch in scripts, which does create the composite. If everything works fine for you, you can approve this PR.

mmaelicke commented:
I am just realizing, that there are a lot of NaNs in the DataFrame. Does this come from the slightly different timestamps? @AlexDo1  do you have an idea?

mmaelicke commented:
And the tests are failing for the exisiting composite dataset tests. I'll have to check that

AlexDo1 commented:
Really good I looked at the data again.
I had accidentally selected `T_begin` instead of `T_end` as `tstamp` for the Eddy record. This caused the eddy data to be 30 minutes off compared to the other records, I fixed that.

The number of NaN values is very high in this data set. I have replaced all values < -9999 with NaN before uploading.

Raw data: number of values < -9999 is `1528364`, number of NaN is `285` --> `1528649`
Uploaded Metacatalog data: number of NaN: `1528649`

mmaelicke commented:
> Really good I looked at the data again.
> I had accidentally selected `T_begin` instead of `T_end` as `tstamp` for the Eddy record. This caused the eddy data to be 30 minutes off compared to the other records, I fixed that.
> 
Is that fixed in the scripts repo? So with the next upload it will work as expected?

> The number of NaN values is very high in this data set. I have replaced all values < -9999 with NaN before uploading.
> 
> Raw data: number of values < -9999 is `1528364`, number of NaN is `285` --> `1528649`
> Uploaded Metacatalog data: number of NaN: `1528649`
Alright. Then, if the image above looks good to you, I guess I will sort out the failing tests and merge this PR afterwards...
Thank you Alex

AlexDo1 commented:
> > Really good I looked at the data again.
> > I had accidentally selected `T_begin` instead of `T_end` as `tstamp` for the Eddy record. This caused the eddy data to be 30 minutes off compared to the other records, I fixed that.
> 
> Is that fixed in the scripts repo? So with the next upload it will work as expected?

This is fixed in the latest commit in the `eddy_upload` branch, yes.
